### PR TITLE
Remove image data constructor to fix release build

### LINKED ISSUE #3222: napi: Fix release build
Building with `napi build --platform --release` produces the following error:

```
`napi` macro expand failed.
error: Constructor has already been defined for struct `JsImageData`
  --> api/napi/src/types/image_data.rs:24:6
   |
24 | impl JsImageData {
   |      ^^^^^^^^^^^
```

The corresponding code declares two constructor functions:

```rust
   #[napi(constructor)]
    pub fn new(width: u32, height: u32) -> Self {
        Self { inner: Image::from_rgba8(SharedPixelBuffer::new(width, height)) }
    }

    // FIXME: constructor with buffer causes trouble
    #[napi(constructor)]
    pub fn from_data_array(data_array: Buffer, width: u32) -> Self {
        Self {
            inner: Image::from_rgba8(SharedPixelBuffer::clone_from_slice(
                data_array.as_ref(),
                width,
                width / data_array.len() as u32,
            )),
        }
    }
```

TypeScript supports function overloads by generating a wrapper that looks at the arguments and calls the right inlined variant. It's unclear how to achieve this with napi.

### PR DESCRIPTION
**Pull request by FloVanGH**
Closes #3222

### CODE REVIEW

File: api/napi/__test__/js_value_conversion.spec.ts [hunk 1]
```diff
@@ -236,7 +236,7 @@ test('get/set image properties', async (t) => {
 
     instance!.invoke("image-width", [0]);
     instance!.invoke("image-height", [0]);
-    instance!.setProperty("alt-image", new ImageData(image.bitmap.data, 64));
+    instance!.setProperty("alt-image", { width: 64, height: 64, data: new Uint8ClampedArray(image.bitmap.data) });
```
Comment (tronical) [inline api/napi/__test__/js_value_conversion.spec.ts:239]: I wonder: What's the difference, testing wise, between this line and line 231? Maybe this entire block isn't needed?

**Review by tronical — COMMENTED** [review]
Please also enable the release build in the CI (as the commit message says) and add a fixes tag to the commit message close the corresponding issue.

(also typo: "contrstructor" -> "constructor" :)

File: api/napi/__test__/js_value_conversion.spec.ts (hunk 1 above)
Comment (FloVanGH) [reply api/napi/__test__/js_value_conversion.spec.ts]: Yes you're right. It's the same I have removed the test.

Comment (tronical) [reply api/napi/__test__/js_value_conversion.spec.ts]: You have removed the line, but not the entire block. I think the test will fail in the CI. You can also remove the corresponding bits in the .slint code if they are not needed anymore.

Comment (FloVanGH) [reply api/napi/__test__/js_value_conversion.spec.ts]: is removed

File: api/napi/__test__/js_value_conversion.spec.ts [hunk 2]
```diff
@@ -172,20 +172,8 @@ test('get/set image properties', async (t) => {
   let compiler = new ComponentCompiler;
   let definition = compiler.buildFromSource(`
   export component App {
-    callback image-width() -> float;
-    callback image-height() -> float;
-
     in-out property <image> image: @image-url("resources/rgb.png");
     in property <image> alt-image;
```
Comment (tronical) [inline api/napi/__test__/js_value_conversion.spec.ts:176]:
```suggestion
```

**Review by tronical — APPROVED** [review]

**Pull request merged**